core/src/window/id.rs

```diff
@@ -0,0 +1,21 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+/// The id of the window.
+///
+/// Internally Iced reserves `window::Id::MAIN` for the first window spawned.
+pub struct Id(u64);
+
+impl Id {
+    /// The reserved window [`Id`] for the first window in an Iced application.
+    pub const MAIN: Self = Id(0);
+
+    /// Creates a new unique window [`Id`].
+    pub fn new(id: impl Hash) -> Id {
+        let mut hasher = DefaultHasher::new();
```

Comment:
I am somewhat concerned about how Iced is using hashes like this. Generally to avoid collisions you'd want a 128-bit hash like MD5 (which still isn't robust against malicious inputs; but that may generally not be an issue for window IDs anyway).

At least for window IDs there are likely to not be very many of them.

Using an incrementing counter would be much more reliable, but wouldn't offer any easy way to statically allocate IDs. (Though 0 could still be reserved for the main window.)

Reply:
Can you think of a use case where this would actually be an issue? iirc the `DefaultHasher` is siphash 1-3, which would need an average of 2^32 hashes before having a collision, unless I'm missing something. I will admit my knowledge of hashing is not the best.

Reply:
Probably it's more a concern with something like the hashing used for image `Handle`s than here. In that case, presumably it would be quite possible for a malicious actor to generate images that collide, which may be an issue in some applications.

Maybe it's okay here.

Reply:
I'm fine with switching to an incrementing internal window ID count as well, but I believe there were some problems when it was originally tried (..? might be misremembering). I can look into it more 👍 Also fine with switching to a 128 bit hashing output if there are security concerns since storage here is pretty negligible.

Reply:
to get away from hashes you would need a generational array that would store the actual data and give a GenerationalIndex. This does work quiet well but it a much different approach from how the hashing version works. In all likely hood the hashing inside of the application shouldn't be susceptible to outside interference, unless They already have full control of your system or for WASM its loading some things from other websites that could become compromised. 

Hashing can have overlaps but based on how many and how the hashes are being stored and at what representation level. This in a UI library should not have to many issues. Though moving to a GenerationalIndex would solve some other types of issues but would lack in the "reuse of" issues as you would still need to Hash things in order to know if it already exists or not. 

Now if your worried about a bad actor Getting your installer and implementing Compromised images and then re-uploading for people to download. there is not much we can do here since they can reverse engineer and change aspect of the software And any image they do replace will just be as if it exists since we don't Store the hashes until runtime when we load everything. Even if we did he could just replace said stores with reverse engineering the source.